src/main/resources/scripts/cleanup/non_prod_primary_and_replica_data_cleanup.sql

```diff
@@ -0,0 +1,82 @@
+--
+-- RWA-2482 cleanup old tasks from primary task database
+--
+
+CREATE OR REPLACE FUNCTION task_cleanup_primary()
+RETURNS VOID
+LANGUAGE plpgsql
+AS $function$
+DECLARE
+    id text;
+BEGIN
+    -- Loop through the rows of the tasks table
+    FOR id IN select task_id from cft_task_db.tasks WHERE EXTRACT(YEAR FROM created) in (2021) LOOP
+
+        -- delete from sensitive_task_event_logs
+        delete from cft_task_db.sensitive_task_event_logs where task_id = id;
+
+        IF EXISTS (SELECT 1 FROM cft_task_db.sensitive_task_event_logs where task_id = id) THEN
+            RAISE NOTICE 'Issue with the taskId: % in sensitive_task_event_logs table', id;
+        END IF;
+
+        -- delete from task_roles
+        delete from cft_task_db.task_roles where task_id = id;
+
+        IF EXISTS (SELECT 1 FROM cft_task_db.task_roles where task_id = id) THEN
+            RAISE NOTICE 'Issue with the taskId: % in task_roles table', id;
+        END IF;
+
+        -- delete from tasks
+        DELETE FROM cft_task_db.tasks WHERE task_id = id;
+
+        IF EXISTS (SELECT 1 FROM cft_task_db.tasks WHERE task_id = id) THEN
+            RAISE NOTICE 'Issue with the taskId: % in tasks table', id;
+            RETURN;
+        END IF;
+    END LOOP;
+END;
+$function$
+
+--
+-- RWA-2482 cleanup old tasks from replica task database
+--
+
+CREATE OR REPLACE FUNCTION task_cleanup_replica()
+RETURNS VOID
+LANGUAGE plpgsql
+AS $function$
+DECLARE
+    id text;
+BEGIN
+    -- Loop through the rows of the tasks table
+    FOR id IN select task_id from cft_task_db.tasks WHERE EXTRACT(YEAR FROM created) in (2021) LOOP
+
+        -- Check if any more rows exist in tasks
+        IF EXISTS (SELECT 1 FROM cft_task_db.tasks where task_id = id) THEN
+            RAISE NOTICE 'Issue with the taskId: % in tasks table', id;
+        END IF;
```

Comment:
I made the changes accordingly. Since the tasks are being deleted through triggers, Instead looking for tasks  in the replica task table, i am taking distinct tasks from the task_history table.

Reply:
I made changes to skip the task deletion from child tables if the tasks is till there in the replica task table.